### PR TITLE
Business details frontend, part 2. Add a review feature, links to order, google map.

### PR DESCRIPTION
**Pull request by saraholiviawu**
**Key features completed**: 
 - **Add a review feature**: 
    - User can click 'Add a review' button to generate a pop up form
    - Pop up form contains an easy visual for users to 1) rate by number of stars corresponding to experience 2) rate by number of  dollar signs corresponding to price and 3) write comments.
    - Pop up stores user input in local storage and clear when on a page refresh, which occurs user clicks 'Post'
    - Pop up form 'Post' button is disabled until user rates number of stars 
  - **Google map**: Displayed a set location based on hard coded coordinates
  **Display average star ratings & price ratings**: Will be calculated based on all user reviews
  - **Add safety health icon for pickup**
  - **Personal website / Order links**: Add links to business' personal web page (if applicable) and links to order from Grubhub, Postmates, and Uber Eats.

**Generalized function to clear local storage**:
  - takes in an array of keys and removes each value from local storage

**Future steps**:
  -  **Frontend**: 
    - Change color of stars and dollar signs when user hovers over those areas in the pop up form 
    - Add 'reviews section' to see all user's ratings and reviews
    - Add another safety health icon to correspond with indoor dining
     
  -  **Backend**: 
    - Read all business details from Firestore     
    - Write user's review data to Firestore
    - Read all user reviews from Firestore and display these dynamically in a 'reviews section'
    - Calculate average price rating and star rating based on user reviews 
    - Calculate walking directions to restuarant from user's current location

**Screenshots**:
businessdetails_page_1
businessdetails_page_2

**Demo**
1. Add a review feature. Note that the 'Post' button is only enabled once user rates number of stars. Will add hover color response.
See full video: https://drive.google.com/file/d/19uNZBzIIelHgpbjVQ6mJBA0uATo6ZXfR/view
Business details_ Add a review (no hover)

2. Working links to restuarant's personal webpage, Ubereats, Postmates, Grubhub.
See full video: https://drive.google.com/file/d/1KvqZ3kDC8cnw1nnAIGA-uMnjOBL01ibK/view
Business details links

### CODE REVIEW

File: ripple/src/main/webapp/businessdetails.js [hunk 1]
```diff
@@ -1,7 +1,63 @@
 /* When user clicks on a team member, add highlight to that avatar */
-$(document).ready(function(){
+$(document).ready(() => {
   $(".bd-avatar").click(function() {
```
Comment (peter14f) [inline ripple/src/main/webapp/businessdetails.js:3]: Can this be an inline function?

File: ripple/src/main/webapp/businessdetails.js [hunk 2]
```diff
@@ -1,7 +1,63 @@
...
-    $(this).addClass('bd-team-highlight');
-   });
-  });
\ No newline at end of file
+    $(this).addClass("bd-team-highlight");
+  });
+});
+
+$(".selector").on("mouseover", function () {
+    //stuff to do on mouseover
```
Comment (peter14f) [inline ripple/src/main/webapp/businessdetails.js:10]: Can this be an inline function?

File: ripple/src/main/webapp/businessdetails.html [hunk 3]
```diff
@@ -89,16 +89,88 @@
...
+      <button type="button" id="review-button" class="primary-fill-chip chip-box-shadow bd-subheader-padding-top" data-toggle="modal" data-target="#bdModal">
+        Add a review
+      </button>
+      <!-- The Modal -->
+      <div class="modal show fade" id="bdModal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true">
+        <div class="modal-dialog">
+          <div class="modal-content">
+            <!-- Modal header -->
+            <div class="modal-header center">
+              <h3 class="col-12 modal-title bd-center-align">
```
Comment (smruthibalajee) [inline ripple/src/main/webapp/businessdetails.html:125]: Add more padding above the write review text and on the sides

File: ripple/src/main/webapp/businessdetails.html [hunk 4]
```diff
@@ -89,16 +89,88 @@
...
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                  <span aria-hidden="true">&times;</span>
+                </button>
+              </h3>
+            </div>
+            <!--/Modal  -->
+            <!-- Modal body -->
+            <div class="modal-body">
+              <div class="bd-review-subwrapper">
+                <!-- User rates stars -->
```
Comment (smruthibalajee) [inline ripple/src/main/webapp/businessdetails.html:136]: Could more padding also be added around the text box and between the star/dollar ratings and text box?

File: ripple/src/main/webapp/businessdetails.js (hunk 1 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js:3]: Fixed document.ready to inline function syntax. Nested function responding to click cannot be an inline function

File: ripple/src/main/webapp/businessdetails.js (hunk 2 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.js:10]: Deleted function itself b/c it is not being used.

File: ripple/src/main/webapp/businessdetails.html (hunk 3 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.html:125]: Fixed. See screenshot below
Write a review V3

File: ripple/src/main/webapp/businessdetails.html (hunk 4 above)
Comment (saraholiviawu) [reply ripple/src/main/webapp/businessdetails.html:136]: Fixed.

**Pull request merged**